homeassistant/components/vlc_telnet/config_flow.py

```diff
@@ -151,9 +152,12 @@ async def async_step_reauth_confirm(
             errors=errors,
         )
 
-    async def async_step_hassio(self, discovery_info: dict[str, Any]) -> FlowResult:
+    async def async_step_hassio(self, discovery_info: Any) -> FlowResult:
         """Handle the discovery step via hassio."""
         await self.async_set_unique_id("hassio")
+        # Temporary check whilst the migration to HassioServiceInfo is in progress
+        if isinstance(discovery_info, HassioServiceInfo):
```

Comment:
Why don't we know what is passed here?

Reply:
I was adjusting the all the tests first, and then the code in a second pass after I have adjusted the signature:
https://github.com/home-assistant/core/blob/272b14755091ab85abd5c1aeb00b7ce52928c75c/homeassistant/components/hassio/discovery.py#L123

Reply:
But then the tests don't reflect reality.

Reply:
It reflects what it is about to be.

Reply:
Start with the source and make it backwards compatible. Then adjust outwards, one integration at a time.

Reply:
I'll adjust in #60824